test/testhelper/client/k8s/client.go

```diff
@@ -0,0 +1,276 @@
+// Copyright © 2023 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sealerio/sealer/pkg/checker"
+	"github.com/sealerio/sealer/test/testhelper"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	ReadyStatus = "Ready"
+	TRUE        = "True"
+	FALSE       = "False"
+)
+
+type Client struct {
+	client *kubernetes.Clientset
+}
+
+type NamespacePod struct {
+	Namespace v1.Namespace
+	PodList   *v1.PodList
+}
+
+type EventPod struct {
+	Reason    string
+	Message   string
+	Count     int32
+	Type      string
+	Action    string
+	Namespace string
+}
+
+func NewK8sClient(sshClient *testhelper.SSHClient) (*Client, error) {
+	kubeconfigPath := filepath.Join("/root", ".kube", "config")
+
+	data := testhelper.GetFileDataRemote(sshClient, kubeconfigPath)
+	data = bytes.ReplaceAll(data, []byte("apiserver.cluster.local"), []byte(sshClient.RemoteHostIP.String()))
```

Comment:
Please do not rely on the `cluster.local` domain which is configurable. 

Maybe it's better to  replace `server: https://(.*):6443` to `server: https://sshClient.RemoteHostIP.String():6443`